src/goto-symex/symex_clean_expr.cpp

```diff
@@ -135,12 +135,26 @@ static void adjust_byte_extract_rec(exprt &expr, const namespacet &ns)
   }
 }
 
+static void
+replace_nondet(exprt &expr, symex_nondet_generatort &build_symex_nondet)
+{
+  if(expr.id() == ID_side_effect && expr.get(ID_statement) == ID_nondet)
+  {
+    nondet_symbol_exprt new_expr = build_symex_nondet(expr.type());
+    new_expr.add_source_location() = expr.source_location();
+    expr.swap(new_expr);
+  }
+  else
```

Comment:
Braces would be appreciated around a multiline block

Reply:
Now added